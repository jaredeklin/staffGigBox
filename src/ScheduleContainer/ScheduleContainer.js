import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Schedule } from '../Schedule/Schedule';

const ScheduleContainer = ({
  editSchedule,
  deleteFromSchedule,
  scheduleGenerator,
  unscheduledEvents,
  admin,
  staff,
  schedule,
  currentUser,
  location,
  api
}) => {
  const correctSchedule = api.getIndividualSchedules(
    location.pathname,
    schedule,
    unscheduledEvents,
    currentUser
  );

  const events = correctSchedule.map(event => {
    return (
      <Schedule
        editSchedule={editSchedule}
        staff={staff}
        event={event}
        key={event.event_id}
        deleteFromSchedule={deleteFromSchedule}
        admin={admin}
      />
    );
  });

  return (
    <div>
      {location.pathname.includes('unscheduled') && (
        <div>
          {!unscheduledEvents.length && <h4>There are no unscheduledEvents</h4>}
          <h4>Would you like to fill all unscheduled events?</h4>
          <button className="generate-schedule-btn" onClick={scheduleGenerator}>
            Generate schedule
          </button>
        </div>
      )}
      {events}
    </div>
  );
};

ScheduleContainer.propTypes = {
  editSchedule: PropTypes.func,
  staff: PropTypes.arrayOf(PropTypes.object),
  event: PropTypes.arrayOf(PropTypes.object),
  schedule: PropTypes.arrayOf(PropTypes.object),
  unscheduledEvents: PropTypes.arrayOf(PropTypes.object),
  deleteFromSchedule: PropTypes.func,
  admin: PropTypes.bool,
  scheduleGenerator: PropTypes.func,
  scheduleType: PropTypes.string
};

export default withRouter(ScheduleContainer);
