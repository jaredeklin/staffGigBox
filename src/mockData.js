export const mockStaff = [
  {
    staff_id: 5,
    name: 'TK',
    bartender: true,
    barback: false,
    bar_manager: true,
    ass_bar_manager: true,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1'
  },
  {
    staff_id: 2,
    name: 'Jared',
    bartender: true,
    barback: false,
    bar_manager: false,
    ass_bar_manager: true,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN2'
  },
  {
    staff_id: 3,
    name: 'Jesse',
    bartender: true,
    barback: false,
    bar_manager: false,
    ass_bar_manager: false,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1'
  },
  {
    staff_id: 4,
    name: 'Ross',
    bartender: false,
    barback: true,
    bar_manager: false,
    ass_bar_manager: false,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN2'
  }
];

export const expectedStaff = [
  {
    id: 5,
    name: 'TK',
    bartender: true,
    barback: false,
    bar_manager: true,
    ass_bar_manager: true,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1'
  }
];

export const expectedStaffRoles = [
  { event_id: 3, role: 'Bar Manager', staff_id: 5 },
  { event_id: 3, role: 'Assistant Bar Manager', staff_id: 2 },
  { event_id: 3, role: 'Bartender', staff_id: 3 },
  { event_id: 3, role: 'Barback', staff_id: 4 }
];

export const mockUnscheduleStaff = [
  { event_id: 3, role: 'Bar Manager', staff_id: null, name: 'Staff Needed' },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: null,
    name: 'Staff Needed'
  },
  { event_id: 3, role: 'Bartender', staff_id: null, name: 'Staff Needed' },
  { event_id: 3, role: 'Barback', staff_id: null, name: 'Staff Needed' }
];

export const mockEmptySchedule = [
  {
    event_id: 3,
    role: 'Bar Manager',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 1
  },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 2
  },
  {
    event_id: 3,
    role: 'Bartender',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 3
  },
  {
    event_id: 3,
    role: 'Barback',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 5
  }
];

export const mockEmptyScheduleWName = [
  {
    event_id: 3,
    role: 'Bar Manager',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 1
  },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 2
  },
  {
    event_id: 3,
    role: 'Bartender',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 3
  },
  {
    event_id: 3,
    role: 'Barback',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 5
  }
];

export const mockEventInfo = {
  ass_bar_manager: true,
  bar_manager: true,
  barbacks: 1,
  bartenders: 1,
  date: 'Jul 20, 2018',
  event_id: 3,
  name: 'Billy Prince Billy'
};

export const mockEvent2Info = {
  ass_bar_manager: true,
  bar_manager: true,
  barbacks: 1,
  bartenders: 1,
  date: 'Jul 20, 2018',
  event_id: 4,
  name: 'Not Billy Prince Billy'
};

export const mockUnscheduledEvent = {
  ass_bar_manager: true,
  bar_manager: true,
  barbacks: 1,
  bartenders: 1,
  date: 'Jul 20, 2018',
  event_id: 3,
  name: 'Billy Prince Billy',
  staff: mockEmptyScheduleWName
};

export const mockEventsInfo = [mockEventInfo, mockEvent2Info];

export const mockSchedule = [
  { ...mockEventInfo, staff: mockEmptyScheduleWName }
];

export const mockFEScheduleData = [{ ...mockEventInfo, staff: mockStaff }];

export const mockFillRolesReturn = [
  { event_id: 3, role: 'Bar Manager', staff_id: 5, name: 'TK' },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: 2,
    name: 'Jared'
  },
  { event_id: 3, role: 'Barback', staff_id: 4, name: 'Ross' },
  { event_id: 3, role: 'Bartender', staff_id: 3, name: 'Jesse' }
];
////////////////
export const mockUnscheduledEvents = [
  { ...mockEventInfo, staff: mockUnscheduleStaff },
  { ...mockEvent2Info, staff: mockUnscheduleStaff }
];

export const mockNewSchedule = [
  { ...mockEventInfo, staff: mockFillRolesReturn },
  { ...mockEvent2Info, staff: [] }
];

export const mockEventToFillRoles = [
  {
    ...mockEventInfo,
    staff: [
      {
        event_id: 3,
        role: 'Bar Manager',
        staff_id: null,
        name: 'Staff Needed'
      },
      {
        event_id: 3,
        role: 'Assistant Bar Manager',
        staff_id: null,
        name: 'Staff Needed'
      },
      { event_id: 3, role: 'Bartender', staff_id: 3, name: 'Jesse' },
      { event_id: 3, role: 'Barback', staff_id: null, name: 'Staff Needed' }
    ]
  }
];

export const mockBuildRolesReturn = [
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bar Manager'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Assistant Bar Manager'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback'
  }
];

export const mockPostScheduleReturn = [
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bar Manager',
    schedule_id: 1
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Assistant Bar Manager',
    schedule_id: 2
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 3
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 4
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 5
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 6
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback',
    schedule_id: 7
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback',
    schedule_id: 8
  }
];

export const mockApiSchedule = [
  { schedule_id: 1, event_id: 3, staff_id: 3, role: 'Bartender' },
  { schedule_id: 2, event_id: 3, staff_id: 4, role: 'Barback' },
  { schedule_id: 3, event_id: 3, staff_id: null, role: 'Bar Manager' },
  {
    schedule_id: 4,
    event_id: 3,
    staff_id: null,
    role: 'Assistant Bar Manager'
  },
  { schedule_id: 5, event_id: 4, staff_id: 3, role: 'Bartender' }
];

export const mockApiScheduleReturn = {
  schedule: [],
  unscheduledEvents: [
    {
      ass_bar_manager: true,
      bar_manager: true,
      barbacks: 1,
      bartenders: 1,
      date: 'Jul 20, 2018',
      event_id: 3,
      name: 'Billy Prince Billy',
      staff: [
        {
          staff_id: 3,
          name: 'Jesse',
          bartender: true,
          barback: false,
          bar_manager: false,
          ass_bar_manager: false,
          beer_bucket: false,
          google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1',
          event_id: 3,
          role: 'Bartender',
          schedule_id: 1
        },
        {
          staff_id: 4,
          name: 'Ross',
          bartender: false,
          barback: true,
          bar_manager: false,
          ass_bar_manager: false,
          beer_bucket: false,
          google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN2',
          event_id: 3,
          role: 'Barback',
          schedule_id: 2
        },
        {
          staff_id: null,
          name: 'Staff Needed',
          event_id: 3,
          role: 'Bar Manager',
          schedule_id: 3
        },
        {
          staff_id: null,
          name: 'Staff Needed',
          event_id: 3,
          role: 'Assistant Bar Manager',
          schedule_id: 4
        }
      ]
    }
  ]
};
