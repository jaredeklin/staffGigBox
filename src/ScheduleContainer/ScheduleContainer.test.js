import React from 'react';
import { shallow } from 'enzyme';
import ScheduleContainer from './ScheduleContainer';
import { mockFEScheduleData } from '../mockData';

describe('ScheduleContainer', () => {
  const mockSchedule = [{ ...mockFEScheduleData[0], venue: 'Ogden Theatre' }];
  const mockUnscheduledEvents = [];
  const mockCurrentUser = { staff_id: 3 };
  const mockApi = { getIndividualSchedules: jest.fn(() => mockSchedule) };
  const mockEditSchedule = jest.fn();
  const mockStaff = [];
  const mockDeleteFromSchedule = jest.fn();
  const mockAdmin = true;

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <ScheduleContainer
        schedule={mockSchedule}
        unscheduledEvents={mockUnscheduledEvents}
        location={{ pathname: '/schedule/ogden' }}
        currentUser={mockCurrentUser}
        api={mockApi}
        editSchedule={mockEditSchedule}
        deleteFromSchedule={mockDeleteFromSchedule}
        staff={mockStaff}
        admin={mockAdmin}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  xit('should match the snapshot when there are no unscheduled events', () => {
    const mockLocation = { pathname: '/unscheduled-events' };
    const wrapper = shallow(
      <ScheduleContainer
        schedule={mockSchedule}
        unscheduledEvents={mockUnscheduledEvents}
        location={mockLocation}
        currentUser={mockCurrentUser}
        api={mockApi}
        editSchedule={mockEditSchedule}
        deleteFromSchedule={mockDeleteFromSchedule}
        staff={mockStaff}
        admin={mockAdmin}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
