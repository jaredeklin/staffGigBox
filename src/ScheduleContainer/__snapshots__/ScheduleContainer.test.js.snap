// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(ScheduleContainer)
    admin={true}
    api={
        Object {
            "getIndividualSchedules": [Function],
          }
    }
    currentUser={
        Object {
            "staff_id": 3,
          }
    }
    deleteFromSchedule={[Function]}
    editSchedule={[Function]}
    location={
        Object {
            "pathname": "/schedule/ogden",
          }
    }
    schedule={
        Array [
            Object {
              "ass_bar_manager": true,
              "bar_manager": true,
              "barbacks": 1,
              "bartenders": 1,
              "date": "Jul 20, 2018",
              "event_id": 3,
              "name": "Billy Prince Billy",
              "staff": Array [
                Object {
                  "ass_bar_manager": true,
                  "bar_manager": true,
                  "barback": false,
                  "bartender": true,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN1",
                  "name": "TK",
                  "staff_id": 5,
                },
                Object {
                  "ass_bar_manager": true,
                  "bar_manager": false,
                  "barback": false,
                  "bartender": true,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN2",
                  "name": "Jared",
                  "staff_id": 2,
                },
                Object {
                  "ass_bar_manager": false,
                  "bar_manager": false,
                  "barback": false,
                  "bartender": true,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN1",
                  "name": "Jesse",
                  "staff_id": 3,
                },
                Object {
                  "ass_bar_manager": false,
                  "bar_manager": false,
                  "barback": true,
                  "bartender": false,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN2",
                  "name": "Ross",
                  "staff_id": 4,
                },
              ],
              "venue": "Ogden Theatre",
            },
          ]
    }
    staff={Array []}
    unscheduledEvents={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ScheduleContainer should match the snapshot when there are no unscheduled events 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(ScheduleContainer)
    admin={true}
    api={
        Object {
            "getIndividualSchedules": [Function],
          }
    }
    currentUser={
        Object {
            "staff_id": 3,
          }
    }
    deleteFromSchedule={[Function]}
    editSchedule={[Function]}
    event={
        Object {
            "event_id": 1,
          }
    }
    location={
        Object {
            "pathname": "/unscheduled-events",
          }
    }
    schedule={
        Array [
            Object {
              "ass_bar_manager": true,
              "bar_manager": true,
              "barbacks": 1,
              "bartenders": 1,
              "date": "Jul 20, 2018",
              "event_id": 3,
              "name": "Billy Prince Billy",
              "staff": Array [
                Object {
                  "ass_bar_manager": true,
                  "bar_manager": true,
                  "barback": false,
                  "bartender": true,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN1",
                  "name": "TK",
                  "staff_id": 5,
                },
                Object {
                  "ass_bar_manager": true,
                  "bar_manager": false,
                  "barback": false,
                  "bartender": true,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN2",
                  "name": "Jared",
                  "staff_id": 2,
                },
                Object {
                  "ass_bar_manager": false,
                  "bar_manager": false,
                  "barback": false,
                  "bartender": true,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN1",
                  "name": "Jesse",
                  "staff_id": 3,
                },
                Object {
                  "ass_bar_manager": false,
                  "bar_manager": false,
                  "barback": true,
                  "bartender": false,
                  "beer_bucket": false,
                  "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN2",
                  "name": "Ross",
                  "staff_id": 4,
                },
              ],
              "venue": "Ogden Theatre",
            },
          ]
    }
    staff={Array []}
    unscheduledEvents={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
